### PR TITLE
Make gnidan happy - Add airdrops detection

### PR DESCRIPTION
**Pull request by LefterisJP**
Make @gnidan happy: https://twitter.com/gnidan/status/1342988502917591044

We still need:

- [x] @kelsos to add a page to show those in the frontend
- [ ] Figure out where $TORN is. I did not see it in coingecko. Not released yet?
- [ ] Get more of those lists. More airdrops, for things like $CORN, $GRAIN etc (perhaps @banteg can help here?)
- [ ] Perhaps for premium users detect if they are already claimed or not. If not, ping them to do so.

----

@gnidan you can still use/test it without the frontend.

Just query the backend like this (it's a rest API): `http GET http://localhost:5042/api/1/blockchains/ETH/airdrops` and you will get a mapping of any of your tracked addresses and what they are entitled to where

(the port may differ -- depends on how you start rotki)

### CODE REVIEW

**Comment by LefterisJP** [comment]
Actually there is even an issue for this: https://github.com/rotki/rotki/issues/972

**Comment by kelsos** [comment]
@LefterisJP would it make sense to have an airdrops section in the dashboard + adding them under each account?

**Comment by LefterisJP** [comment]
@kelsos dashboard probably not. But adding them under each account yes.

File: frontend/app/src/assets/images/1inch.svg [hunk 1]
```diff
@@ -0,0 +1,183 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (LefterisJP) [inline frontend/app/src/assets/images/1inch.svg:1]: you took their christmas SVG version. May need to change it later :stuck_out_tongue:

File: frontend/app/src/views/defi/Airdrops.vue [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    const addresses = this.selectedAccounts.map(({ address }) => address);
+    return this.airdrops(addresses);
+  }
+
+  async refresh() {
+    await this.fetchAirdrops(true);
+  }
+
+  readonly sources: AirdropSources = {
+    [AIRDROP_UNISWAP]: {
```
Comment (LefterisJP) [inline frontend/app/src/views/defi/Airdrops.vue:129]: So when we add more, we need to edit those? Need to find a way to keep those in sync in between frontend and backend. Just like with the DeFi protocols too.

**Review by LefterisJP — COMMENTED** [review]
Okay frontend works. Just some minor comments which probably do not need action on the frontend yet.

The only thing I would like you to add now is a text here above the filter:




Explaining what this section does at the moment. So something like:

> This is the airdrops Rotki detected your tracked accounts are eligible for. You may have already claimed them. If not then visit the link for each protocol and claim your share.

And I can make the backend also return a link per protocol? So you can add it in the table, and users can easily click and go to the project's page. Some airdrops like tornado did a passive airdrop. So it has automatically been distributed already.

File: frontend/app/src/assets/images/1inch.svg (hunk 1 above)
Comment (kelsos) [reply frontend/app/src/assets/images/1inch.svg:1]: Well it is was they have at their site :P

File: frontend/app/src/views/defi/Airdrops.vue (hunk 2 above)
Comment (kelsos) [reply frontend/app/src/views/defi/Airdrops.vue:129]: Well basically this is what we do for the defi protocols too :P

Comment (LefterisJP) [reply frontend/app/src/views/defi/Airdrops.vue:129]: yeah I know but we need to figure out a better way for all these stuff to be able to make changes in one place, or at least detect when they fall out of sync. Else it won't scale. Can you spend some thinking juice on it? Probably the test way would make sense?

Comment (kelsos) [reply frontend/app/src/views/defi/Airdrops.vue:129]: Well one way would be to move this metadata and icon serving to the backend. no need to maintain it twice then.

Comment (LefterisJP) [reply frontend/app/src/views/defi/Airdrops.vue:129]: As discussed in the standup can you make an issue for this @kelsos? PErhaps you can also pick it up but we will see. This is not for v1.11.0 though!

File: frontend/app/src/assets/images/1inch.svg (hunk 1 above)
Comment (LefterisJP) [reply frontend/app/src/assets/images/1inch.svg:1]: Season to be jolly :)

Comment (kelsos) [reply frontend/app/src/assets/images/1inch.svg:1]: When Christmas rotki? :P

**Pull request merged**